### PR TITLE
[CI] Add release workflow

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit adds release GitHub Actions workflow.
It checks license, copyright issues and
runs build test and unit tests. There is no issue for
release, it creates vsce package and release it.

ONE-vscode-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

Related issue: #571 

Draft workflow: https://github.com/jyoungyun/ONE-vscode/actions/runs/2298576642

This workflow made an issue for release. ([example](https://github.com/jyoungyun/ONE-vscode/issues/7))
It runs several release tests and if everything is successful, it releases new version. ([example](https://github.com/jyoungyun/ONE-vscode/releases/tag/0.0.1))

### CODE REVIEW

**Review by seanshpark — APPROVED** [review]
LGTM

File: .github/workflows/build-release.yml [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    inputs:
+      tag:
+        description: 'Release tag (format: major.minor.patch)'
+        required: true
+      draft:
+        description: 'Create a draft (unpublished) release (default: false)'
+        type: boolean
+        default: false
+      prerelease:
+        description: 'Identify the release as a prerelease (default: false)'
```
Comment (hyunsik-yoon) [inline .github/workflows/build-release.yml:14]: I guess `draft` in draft release is similar with `draft` in draft PR.
For pre-release, I found https://semver.org/#spec-item-9.
Is pre-release for something like alpha version or beta version?

**Review by hyunsik-yoon — APPROVED** [review]

File: .github/workflows/build-release.yml (hunk 1 above)
Comment (jyoungyun) [reply .github/workflows/build-release.yml:14]: These features are related to GitHub Release system.

When you release a new version in GitHub System, you can use `draft` and `prerelease` items.

Pre-release in GitHub Release means that the release is not ready for production and may be unstable.
Draft in GitHub Release means that if you're ready to publicize your release but to work on the release later, use `draft` release.

File: .github/workflows/build-release.yml [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v3
+        with:
+          name: ${{ needs.create-vsce-package.outputs.package_name }}
+      - name: Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (llFreetimell) [inline .github/workflows/build-release.yml:235]: Did we add this in the repository setting?

Comment (jyoungyun) [reply .github/workflows/build-release.yml:235]: No. This token is automatically created. :)
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret

File: .github/workflows/build-release.yml [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+              issue_number: ${{ needs.create-issue.outputs.issue_number }},
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: body_content
+            });
+  delete-license-result:
+    needs: post-license-result
+    runs-on: ubuntu-latest
+    steps:
+      - uses: geekyeggo/delete-artifact@v1
```
Comment (llFreetimell) [inline .github/workflows/build-release.yml:95]: We may be able to substitute this with our own code later :)

**Review by llFreetimell — APPROVED** [review]
LGTM

**Pull request merged**